<launch>

  <!-- I don't think we need this if there is a graphics card-->
  <!-- <env name="LIBGL_ALWAYS_SOFTWARE" value="1"/ -->

  <!-- read value of arg -->
  <param name="param" value="$(arg lidar_file)"/>
  <param name="param" value="$(arg cad_file)"/>
  <param name="param" value="$(arg thresh)"/>

  <node
    pkg="seam_detection"
    name="setup_tf"
    type="setup_tf"
    output="screen"
  />

  <!--
  <node
    pkg="pcl_ros"
    name="pcd_to_pointcloud"
    type="pcd_to_pointcloud"
    output="screen"
    args="$(find seam_detection)/images/$(arg lidar_file)">
  </node>
  -->

  <node name="rviz"
        pkg="rviz"
        type="rviz"
        args="-d $(find seam_detection)/rviz/seam_detection_ICP.rviz">
  </node>

  <node
    pkg="seam_detection"
    name="seam_detection_ICP"
    type="seam_detection_ICP"
    output="screen"
    args="-f $(find seam_detection)/pcd_images/$(arg lidar_file) $(find seam_detection)/pcd_images/$(arg cad_file) $(arg thresh)"
  />

</launch>
